### PR TITLE
Use Jenkins Agent image from Digital Grid Artifactory

### PR DESCRIPTION
**Pull request by Yogeshkumarb57**

### CODE REVIEW

**Review by johnson-jeb — COMMENTED** [review]
Still references are in the DocsJenkinsfile, do we need to fix here also? or is it unused jenkins file?

**Comment by Yogeshkumarb57** [comment]
> Still references are in the DocsJenkinsfile, do we need to fix here also? or is it unused jenkins file?

Not sure about usage of DocsJenkinsfile but it is using old "uaa-ci-testing:1.1.0" image [https://github.com/GESoftware-CF/uaa/blob/predix_extensions_75.18.1/DocsJenkinsfile#L3-L4, https://github.build.ge.com/predix/security-ci-commons/blob/master/vars/nodeDetails.groovy#L23-L28]

**Review by johnson-jeb — APPROVED** [review]

File: DeployJenkinsfile [hunk 1]
```diff
@@ -1,8 +1,9 @@
 #!/usr/bin/env groovy
 def artifactoryServer = Artifactory.server('Digital-Artifactory')
 
-library "security-ci-commons-shared-lib"
-def NODE = nodeDetails("uaa-upgrade")
+def NODE = [LABEL: "dind", IMAGE: "dig-grid-artifactory.apps.ge.com/pgog-fss-iam-uaa-docker-stage/uaa-ci-testing:latest",
```
Comment (irinaepshteyn) [inline DeployJenkinsfile:4]: Lets specify node details inline in the agent section. Pipeline will read easier

File: DockerizeJenkinsfile [hunk 2]
```diff
@@ -7,7 +7,7 @@ def use_release_artifacts() {
 pipeline {
     agent {
         docker {
-            image 'registry.gear.ge.com/dig-predix-security-services/uaa-ci-testing:latest'
+            image 'dig-grid-artifactory.apps.ge.com/pgog-fss-iam-uaa-docker-stage/uaa-ci-testing:latest'
```
Comment (irinaepshteyn) [inline DockerizeJenkinsfile:10]: This file and corresponding pipeline are outdated and not used any more. Let's leave this file as is.

File: Jenkinsfile [hunk 3]
```diff
@@ -1,8 +1,9 @@
 #!/usr/bin/env groovy
 def artifactoryServer = Artifactory.server('Digital-Artifactory')
 
-@Library(['PPCmanifest','security-ci-commons-shared-lib']) _
-def NODE = nodeDetails("uaa-upgrade")
+@Library(['PPCmanifest']) _
```
Comment (irinaepshteyn) [inline Jenkinsfile:4]: This library is not used anymore; let's remove. It was used in the past to publish build artifacts to BinTray for onprem Rosneft customer. The stage to publish to BinTray has been removed, so no need for the library.

**Comment by irinaepshteyn** [comment]
@Yogeshkumarb57, I understand that you modified Jenkins system configuration to add Digital Artifactory as Docker Registry. That seems to overwrite global default registry.gear.ge.com Docker Registry. It made all pipelines to switch to using Digital Artifactory instead of registry.gear.ge.com. As a result, it broke UAA Dashboard that is still using registry.gear.ge.com to pull its agent: "Error response from daemon: unknown: Repository 'registry.gear.ge.com' not found" in [job/develop/23](https://i.ci.build.ge.com/bjmjek1k/ci/job/UAA%20Dashboard/job/UAA%20Dashboard%20Build%20N%27%20Test/job/develop/23/console) 

It will also brake UAA pipelines for rc and release branch that don't have your changes yet. So until we implement the same change for UAA Dashboard and until we merge your changes to all branches of UAA, we need to find a workaround of using both Docker registries. Maybe you could use [withDockerRegistry](https://www.jenkins.io/doc/pipeline/steps/docker-workflow/#withdockerregistry-sets-up-docker-registry-endpoint) instead of changing system configuration?...

File: AcceptanceEKSJenkinsfile [hunk 4]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env groovy
-library "security-ci-commons-shared-lib"
-def NODE = nodeDetails("uaa-upgrade")
+def NODE = [LABEL: "dind", IMAGE: "dig-grid-artifactory.apps.ge.com/pgog-fss-iam-uaa-docker-stage/uaa-ci-testing:latest",
+    ARGS: "-v /var/lib/docker/.gradle:/root/.gradle"]
 
 pipeline {
     agent none
```
[the remark this answers is not in the document]
Comment (Yogeshkumarb57) [reply AcceptanceEKSJenkinsfile]: Done

File: DeployJenkinsfile (hunk 1 above)
Comment (Yogeshkumarb57) [reply DeployJenkinsfile]: Done

File: DockerizeJenkinsfile (hunk 2 above)
Comment (Yogeshkumarb57) [reply DockerizeJenkinsfile]: Done, reverted changes.

File: Jenkinsfile (hunk 3 above)
Comment (Yogeshkumarb57) [reply Jenkinsfile]: Done, removed PPCmanifest lib

**Comment by Yogeshkumarb57** [comment]
> @Yogeshkumarb57, I understand that you modified Jenkins system configuration to add Digital Artifactory as Docker Registry. That seems to overwrite global default registry.gear.ge.com Docker Registry. It made all pipelines to switch to using Digital Artifactory instead of registry.gear.ge.com. As a result, it broke UAA Dashboard that is still using registry.gear.ge.com to pull its agent: "Error response from daemon: unknown: Repository 'registry.gear.ge.com' not found" in [job/develop/23](https://i.ci.build.ge.com/bjmjek1k/ci/job/UAA%20Dashboard/job/UAA%20Dashboard%20Build%20N%27%20Test/job/develop/23/console)
> 
> It will also break UAA pipelines for rc and release branch that don't have your changes yet. So until we implement the same change for UAA Dashboard and until we merge your changes to all branches of UAA, we need to find a workaround of using both Docker registries. Maybe you could use [withDockerRegistry](https://www.jenkins.io/doc/pipeline/steps/docker-workflow/#withdockerregistry-sets-up-docker-registry-endpoint) instead of changing system configuration?...

Removed digital artifactory creds from Jenkins Global Docker Registery and updated pipelines accordingly. I have also tried [withDockerRegistry](https://www.jenkins.io/doc/pipeline/steps/docker-workflow/#withdockerregistry-sets-up-docker-registry-endpoint) at both places agent and stage steps block but no success.

**Review by irinaepshteyn — APPROVED** [review]

**Pull request merged**